Validation/HcalDigis/src/HcalDigisValidation.cc

```diff
@@ -528,17 +532,18 @@ void HcalDigisValidation::analyze(const edm::Event& iEvent, const edm::EventSetu
 
    for (HcalTrigPrimDigiCollection::const_iterator itr = dataTPs->begin(); itr != dataTPs->end(); ++itr) {
      int ieta  = itr->id().ieta();
-     int iphi  = itr->id().iphi();
 
      HcalSubdetector subdet = (HcalSubdetector) 0;
-     if      ( abs(ieta) <= 16 ) subdet = HcalSubdetector::HcalBarrel ;
-     else if ( abs(ieta) <= 28 ) subdet = HcalSubdetector::HcalEndcap ;
-     else if ( abs(ieta) <= 40 ) subdet = HcalSubdetector::HcalForward;
+     if      ( abs(ieta) <= 16 )
+        subdet = HcalSubdetector::HcalBarrel ;
+     else if ( abs(ieta) < tp_geometry->firstHFTower(itr->id().version()) ) 
+        subdet = HcalSubdetector::HcalEndcap ;
+     else if ( abs(ieta) <= 42 )
```

Comment:
The next round of HCAL validation (1x1 TP related) update is in works.


On Thu, 17 Mar 2016, David Lange wrote:

> 
> In Validation/HcalDigis/src/HcalDigisValidation.cc:
> 
> > 
> >       HcalSubdetector subdet = (HcalSubdetector) 0;
> > -     if      ( abs(ieta) <= 16 ) subdet = HcalSubdetector::HcalBarrel ;
> > -     else if ( abs(ieta) <= 28 ) subdet = HcalSubdetector::HcalEndcap ;
> > -     else if ( abs(ieta) <= 40 ) subdet = HcalSubdetector::HcalForward;
> > +     if      ( abs(ieta) <= 16 )
> > +        subdet = HcalSubdetector::HcalBarrel ;
> > +     else if ( abs(ieta) < tp_geometry->firstHFTower(itr->id().version()) ) 
> > +        subdet = HcalSubdetector::HcalEndcap ;
> > +     else if ( abs(ieta) <= 42 )
> 
> ok, now i see why 42... but could you look at getting rid of it as a new pull request?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub[AEx02mflPtTEiuBpJZvp0FjDeqcP4AHDks5puWOHgaJpZM4Hc6h-.gif]
> 
> 
>